os/compression/compress_read.c

```diff
@@ -263,6 +280,180 @@ static off_t compress_read_block(int filfd, uint16_t binary_header_size, FAR uin
 	return nbytes;
 }
 
+/****************************************************************************
+ * Name: compress_update_blockcache_list
+ *
+ * Description:
+ *   Update blockcache list based on whether 'block_number' block in
+ *   blockwise-compressed binary is already cached or not. Also, maintain
+ *   head, tail and most_accessed pointers. Decompress block if not cached.
+ *   tail = most recently cached block (highest priority block for keeping
+ *          cached)
+ *   head = oldest cached block (lowest priority for keeping it cached).
+ *          If a new block needs to be cached, cache to this blockcache element.
+ *   most_accessed = pointer to most accessed block in blockcache list.
+ *          Needed for easy access to most accessed block from comp. binary.
+ *
+ * Returned Value:
+ *   Index in blockcache list where 'block_number' block number from compressed
+ *   binary is cached in uncompressed form. Return Negative value on failure.
+ ****************************************************************************/
+static inline int compress_update_blockcache_list(int block_number, int filfd, uint16_t binary_header_size)
```

Comment:
Any reason to declare this function as inline?

Reply:
Will check if this improves performance. If not, will remove inline.